samples/asymmetricDecrypt.js

```diff
@@ -0,0 +1,55 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+// [START kms_asymmetric_decrypt]
+async function asymmetricDecrypt(
+  projectId = 'your-project-id', // Your GCP projectId
+  keyRingId = 'my-key-ring', // Name of the crypto key's key ring
+  cryptoKeyId = 'my-key', // Name of the crypto key, e.g. "my-key"
+  cryptoKeyVersionId = '1', // Version of the crypto key to use
+  ciphertext = '...' // Base64-encoded ciphertext data to decrypt
+) {
+  // Import the library and create a client
+  const kms = require('@google-cloud/kms');
+  const client = new kms.KeyManagementServiceClient();
+
+  // The location of the crypto key's key ring, e.g. "global"
+  const locationId = 'global';
+
+  // Construct the cyrpto key version ID
+  const name = client.cryptoKeyVersionPath(
+    projectId,
+    locationId,
+    keyRingId,
+    cryptoKeyId,
+    cryptoKeyVersionId
+  );
+
+  // Decode the base64-encoded ciphertext
+  const ciphertextBuffer = Buffer.from(ciphertext, 'base64');
+
+  // Decrypt plaintext using Cloud KMS
+  const [result] = await client.asymmetricDecrypt({
```

Comment:
@fhinkel outlines a slightly different style for async samples in Node.js:

```js
function main(arg1 = 'default_value_one', arg2 = 1234) {
  // [START your_region_tag]
  /**
   * TODO(developer): Uncomment these variables before running the sample.
   */
  // const arg1 = 'default_value_one';
  // const arg2 = 1234;

  // Imports the Google Cloud Some API library
  const {SomeApiClient} = require('@google-cloud/some-api');

  // Instantiates a client
  const someApiClient = new SomeApiClient();

  async function doSomething() {
    ...
  }

  doSomething();
  // [END your_region_tag]
}

main(...process.argv.slice(2));
```

In which there's an outer and inner function, this is so that when samples are embedded externally the code can be copied and pasted (since top level await doesn't work yet).

Reply:
Interesting. I agree with this approach, but I also followed the existing examples. Would you be open to leaving these as-is and then I'll come through and refactor _all_ the samples to use this new format? I worry it would be confusing to users if sames for the same language-product differed, and I don't want to refactor all the existing samples in this PR to keep it focused.

Reply:
Interesting. I agree with @fhinkel's approach, but I also followed the existing examples. Would you be open to leaving these as-is and then I'll come through and refactor all the samples to use this new format? I worry it would be confusing to users if some for the same language-product differed, and I don't want to refactor all the existing samples in this PR to keep it focused.

Reply:
The samples should not throw when you copy and paste them. That's why we use the slightly longer format. I'm OK with being consistent rather than following our guidelines. I'll make a note that this repo needs to be refactored, cc @sofisl if you're interested in taking this on next.